pkg/controller/machine/controller_test.go

```diff
@@ -42,7 +42,13 @@ func machineControllerReconcile(t *testing.T, cs *clientset.Clientset, controlle
 	if _, err := clusterClient.Create(&cluster); err != nil {
 		t.Fatal(err)
 	}
-	defer clusterClient.Delete(cluster.Name, &metav1.DeleteOptions{})
+	defer func () {
```

Comment:
Perhaps have a helper method for cleaning up cluster and call that method from the defer? That way test cluster cleanup is defined in one place.